Src/Base/AMReX_MFIter.H

```diff
@@ -204,9 +205,9 @@ protected:
     std::unique_ptr<Gpu::FuseSafeGuard> gpu_fsg;
 #endif
 
-    static int nextDynamicIndex;
-    static int depth;
-    static int allow_multiple_mfiters;
+    static AMREX_EXPORT int nextDynamicIndex;
+    static AMREX_EXPORT int depth;
+    static AMREX_EXPORT int allow_multiple_mfiters;
```

Comment:
Is that true we have mark all the static variables and functions in classes with AMREX_EXPORT?

Reply:
Note that these are not static variables at file scopes.

Reply:
Only static variables, since they are encapsulated global states of a library.

Static functions don't need that, as they do not carry a state.

Reply:
In general, for modern C++ libs that template a lot one would manually expose only the public classes with such a macro and use `-fvisibility=hidden -fvisibility-inlines-hidden`.

Benefits / Q&A memo I wrote a while back: https://gist.github.com/ax3l/ba17f4bb1edb5885a6bd01f58de4d542

We simplify this process by making everything visible by default, even for Windows, but [this](https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html) does not include those globals variables.